### PR TITLE
fix(stackdriver): pull mesh uid from config

### LINKED ISSUE #17952: Stackdriver mixer adapter does not support user-configured mesh_uid in contextgraph (but does in metrics and logs)
**Bug description**

I want to include a user-specified mesh_uid in the telemetry reported by the Stackdriver mixer adapter to the stackdriver backend. I can do this for metrics and logs, but I cannot do so for contextgraph, which hardcodes a value like this: https://github.com/istio/istio/blob/1.3.3/mixer/adapter/stackdriver/contextgraph/contextgraph.go#L82

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

If I specify a mesh_uid in the yaml that configures the stackdriver adapter, this value should be sent to the contextgraph backend. (I don't care about the exact details of the yaml format.)

If I don't specify a value, then the current behavior should remain.

**Steps to reproduce the bug**

N/A

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

All versions

**How was Istio installed?**

N/A

**Environment where bug was observed (cloud vendor, OS, etc)**

N/A

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR updates the logic for `mesh_uid` calculation for the stackdriver adapter (in particular, `metrics` and `edges` handling).

The prior design was to use the GCE metadata server, but plans change. This now accepts the argument as a config-time param.

Fixes #17952.

Signed-off-by: Douglas Reid <douglas-reid@users.noreply.github.com>

### CODE REVIEW

**Comment by douglas-reid** [comment]
@geeknoid what do I do about this failed `gencheck_istio` ? I ran `make gen` on my local copy and added those files to this commit. I'm not sure how to next proceed.

**Comment by douglas-reid** [comment]
@bianpengyuan @mandarjog can you please take a look? this is needed to properly populate mesh id (and will be needed for a backport to 1.2, 1.3, 1.4).

Big questions: OK to override `mesh_uid` in metrics (or should it only be if that isn't already set)? Should we do something similar for logs?

**Comment by bianpengyuan** [comment]
If we are passing in the mesh_uid from config instead of fetching from metadata server during runtime, why not just config metric labels and log labels directly?

**Comment by douglas-reid** [comment]
@bianpengyuan the idea, as I understood it, was to simplify the number of places where that config had to be set. and we still need a (backwards-compatible) way to set the mesh_uid for edges too (which the template does not include).

File: mixer/adapter/stackdriver/contextgraph/contextgraph.go [hunk 1]
```diff
@@ -74,9 +73,10 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 	env.Logger().Debugf("Proj, zone, cluster, opts: %s,%s,%s,%s",
 		b.projectID, b.zone, b.cluster, opts)
 
-	meshUID := fmt.Sprintf("%s/%s/%s", b.projectID, b.zone, b.cluster)
-	if len(b.meshID) > 0 {
-		meshUID = b.meshID
+	meshUID := b.cfg.MeshUid
+	if len(meshUID) == 0 {
+		// must be a fallback here
+		meshUID = fmt.Sprintf("%s/%s/%s", b.projectID, b.zone, b.cluster)
```
Comment (bianpengyuan) [inline mixer/adapter/stackdriver/contextgraph/contextgraph.go]: Why do we must have a fallback here? I think it makes more sense to set it only when it is explicitly set in handler.

Comment (douglas-reid) [reply mixer/adapter/stackdriver/contextgraph/contextgraph.go]: backwards compat. if we don't specify it in the handler, the intent is for the CG stuff to continue to work as it did before.

Comment (bianpengyuan) [reply mixer/adapter/stackdriver/contextgraph/contextgraph.go]: Ok, let's hold on adding this to log, and only fill it in metric and edge.

**Pull request merged**